gadget/install/install.go

```diff
@@ -508,14 +515,15 @@ func WriteContent(onVolumes map[string]*gadget.Volume, allLaidOutVols map[string
 			// TODO: do we need to consider different
 			// sector sizes for the encrypted/unencrypted
 			// cases here?
-			onDiskStruct, err := applyLayoutToOnDiskStructure(onDiskVol, volStruct.Device, allLaidOutVols, volName)
+			odls, err := applyLayoutToOnDiskStructure(onDiskVol, volStruct.Device, allLaidOutVols, volName)
 			if err != nil {
 				return nil, fmt.Errorf("cannot retrieve on disk info for %q: %v", volStruct.Device, err)
 			}
 
 			device := deviceForMaybeEncryptedVolume(&volStruct, encSetupData)
 			logger.Debugf("writing content on partition %s", device)
-			if err := writePartitionContent(onDiskStruct, device, observer, perfTimings); err != nil {
+			partDisp := roleOrLabelOrName(odls.laidOut.Role, odls.onDisk)
```

Comment:
here I think Role can be unset? so we do need the whole thing

Reply:
Yeah, think so